### PR TITLE
feat: implement testPermission api of organizations

### PR DESCRIPTION
**Pull request by athakor**

### CODE REVIEW

**Review by pmakani — APPROVED** [review]

File: src/main/java/com/google/cloud/resourcemanager/spi/v1beta1/HttpResourceManagerRpc.java [hunk 1]
```diff
@@ -301,4 +301,25 @@ public Policy replacePolicy(String projectId, Policy newPolicy) throws ResourceM
...
+      TestIamPermissionsResponse response =
+          resourceManager
+              .organizations()
+              .testIamPermissions(
+                  resource, new TestIamPermissionsRequest().setPermissions(permissions))
+              .execute();
+      Set<String> permissionsOwned =
+          ImmutableSet.copyOf(firstNonNull(response.getPermissions(), ImmutableList.<String>of()));
+      ImmutableList.Builder<Boolean> answer = ImmutableList.builder();
+      for (String p : permissions) {
```
Comment (rahulKQL) [inline src/main/java/com/google/cloud/resourcemanager/spi/v1beta1/HttpResourceManagerRpc.java]: Please change the temp variable name to something relevant to the block

**Review by rahulKQL — APPROVED** [review]
LGTM

File: src/main/java/com/google/cloud/resourcemanager/ResourceManager.java [hunk 2]
```diff
@@ -337,4 +337,14 @@ public static ProjectListOption fields(ProjectField... fields) {
...
+  /**
+   * Returns permissions that the caller has on the specified Organization.
+   *
+   * @throws ResourceManagerException upon failure
+   * @see <a href=
+   *     "https://cloud.google.com/resource-manager/reference/rest/v1/organizations/testIamPermissions">
+   *     Resource Manager testIamPermissions</a>
+   */
+  List<Boolean> testOrgPermissions(String resource, List<String> permissions);
 }
```
Comment (dmitry-fa) [inline src/main/java/com/google/cloud/resourcemanager/ResourceManager.java:356]: The doc is not clear here: "Returns permissions that the caller has on the specified Organization."
1) what is 'the specified Organization'?  do you mean resource?
2) 'Returns permissions', but method does not return permissions, it returns list of boolean.
3) What is expected in the 'List<String> permissions' parameter? 

I think you need to add
@param resource The resource for which the policy detail is being requested.
@param permissions the set of permissions to check for the resource. Permissions with wildcards (such as '*' or 'storage.*') are not allowed

IMHO, returning List<String> would be much useful here.

**Review by dmitry-fa — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/cloud/resourcemanager/ResourceManager.java (hunk 2 above)
Comment (athakor) [reply src/main/java/com/google/cloud/resourcemanager/ResourceManager.java:356]: Thanks for this but am disagree with your opinion as method itself said that it's only used to test the permission.means it's only check whether this permissions is exists on resource or not.

Comment (dmitry-fa) [reply src/main/java/com/google/cloud/resourcemanager/ResourceManager.java:356]: I doubt that such API is very convenient, but if there are similar method around, that's fine.
Anyway, the doc should be precise. If you believe, that returning list of boolean is what is needed, please update the doc.

File: src/main/java/com/google/cloud/resourcemanager/spi/v1beta1/HttpResourceManagerRpc.java (hunk 1 above)
Comment (athakor) [reply src/main/java/com/google/cloud/resourcemanager/spi/v1beta1/HttpResourceManagerRpc.java]: done

File: src/main/java/com/google/cloud/resourcemanager/ResourceManager.java (hunk 2 above)
Comment (athakor) [reply src/main/java/com/google/cloud/resourcemanager/ResourceManager.java:356]: done java doc updated with proper description.

**Pull request closed without merging**